projects/storyteller/webapp/src/app/components/single-screenshot/single-screenshot.component.ts

```diff
@@ -1,9 +1,59 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { zip } from 'rxjs';
+
+import { Story, Screenshot, StoryList } from '@/core/proto';
+import { FirebaseService } from '@/core/services';
 
 @Component({
   selector: 'app-single-screenshot',
   templateUrl: './single-screenshot.component.html',
   styleUrls: ['./single-screenshot.component.scss']
 })
-export class SingleScreenshotComponent { }
+export class SingleScreenshotComponent implements OnInit {
+  isLoading = true;
+  storyId: string;
+  screenshotFilename: string;
+  author: string;
+  story: Story;
+  screenshot: Screenshot;
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private firebaseService: FirebaseService
+  ) { }
+
+  ngOnInit() {
+    this.storyId = this.activatedRoute.snapshot.params['storyId'];
+    this.screenshotFilename = this.activatedRoute.snapshot.params['screenshot'];
+    this.author = this.activatedRoute.snapshot.params['storyAuthor'];
+
+    zip(
+      this.firebaseService.getUserStories(this.author),
+      this.firebaseService.getUserScreenshot(this.author)
+    ).subscribe(data => {
+      this.getStory(this.storyId, data[0]);
+      this.getScreenshot(this.screenshotFilename, data[1]);
+      this.isLoading = false;
+    });
+  }
+  getStory(storyId: string, storyLists: StoryList[]): void {
```

Comment:
It'd be great if you'll put empty line between methods

Reply:
Done

Reply:
Great! But actually not only here. There's `FirebaseService` for example